boards/posix/nrf52_bsim/Kconfig.board

```diff
@@ -14,6 +14,11 @@ config BOARD_NRF52_BSIM
 	# of it), so that the corresponding driver becomes available (see
 	# dependencies of the ENTROPY_NRF5_RNG option).
 	select HAS_HW_NRF_RNG
+	# Do the same for the CLOCK and POWER peripherals, so that the nrfx
+	# drivers for them can be used.
+	select HAS_HW_NRF_CLOCK
+	select HAS_HW_NRF_POWER
+	select HAS_NRFX
```

Comment:
I wonder if there is a better way than this. If I understand correctly doing this, means both all the real nrfx and the HW models include paths are included. And things work only because of the include order(?).
If so, that is a bit fragile. It would be nicer if we could include selectively only the drivers folder, without pulling the HAL folders from both sides.
@anangl how would it sound to you, for example, to have separate HAS_NRFX_DRIVERS and has HAS_NRFX_HAL?

Reply:
> If I understand correctly doing this, means both all the real nrfx and the HW models include paths are included. And things work only because of the include order(?).
> If so, that is a bit fragile.

Yes, it relies on the include paths order. And the proper order is enforced by this change: https://github.com/zephyrproject-rtos/hal_nordic/pull/51/commits/3932f46487634d9edd475251507b5a21f4e0dc3f#diff-55df9be073e4151e19fc798a7c38f122. So I think it's not that fragile. And it gives you the possibility of replacing only the nrfx files that you need with their versions from ext_NRF52_hw_models and using others, if needed, directly.

> It would be nicer if we could include selectively only the drivers folder, without pulling the HAL folders from both sides.
> @anangl how would it sound to you, for example, to have separate HAS_NRFX_DRIVERS and has HAS_NRFX_HAL?

I don't really see a way of achieving this. There is no separate include path for nrfx HAL files defined, those are included as `<hal/nrf_*.h>` or `<nrfx/hal/nrf_*.h>` (for instance, in `radio.c` in the bluetooth subsystem).